rust/connlib/tunnel/src/client.rs

```diff
@@ -828,16 +848,8 @@ impl ClientState {
             .map(|(_, res)| res.id)
     }
 
-    fn update_system_resolvers(&mut self, new_dns: Vec<IpAddr>) -> bool {
-        if !dns_updated(&self.system_resolvers, &new_dns) {
-            tracing::debug!("Updated dns called but no change to system's resolver");
-            return false;
-        }
-
-        tracing::info!("Found new system resolvers: {new_dns:?}");
+    fn update_system_resolvers(&mut self, new_dns: Vec<IpAddr>) {
         self.system_resolvers = new_dns;
```

Comment:
So `system_resolvers` here can include sentinels, but the sentinels are filtered out before we update the DNS mapping.

It feels odd. Can we have comments clarifying that, or could `system_resolvers` just move up to the object that owns `role_state`?

Reply:
I think it's correct in the sense that "this are the resolvers that the system told it has set". Ideally I think we won't get any sentinel resolvers. But I'll add a clarifying comment.